compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -341,7 +342,7 @@ pub enum FakeReadCause {
     /// FakeRead for that Place outside the closure, in such a case this option would be
     /// Some(closure_def_id).
     /// Otherwise, the value of the optional DefId will be None.
-    ForMatchedPlace(Option<DefId>),
+    ForMatchedPlace(Option<LocalDefId>),
```

Comment:
Should we add a comment explaining why this variant can never be decoded from upstream crate?

Reply:
I'm afraid I don't know...is it because TypeckResults is never used for upstream crates?

Reply:
As far as I can tell, `FakeReadClause` and `AggegateKind` are only used for statements that are removed from MIR between borrowck and encoding to metadata.  As a consequence, they can't appear in metadata, so we can't load a version with a non-local `DefId`.